src/client/datascience/dataScienceSurveyBanner.ts

```diff
@@ -87,102 +111,235 @@ export type ShowBannerWithExpiryTime = {
+    private async disable(monthsTillNextPrompt: number, type: BannerType) {
+        await this.showBannerState[type].updateValue({
             expiry: monthsTillNextPrompt * 31 * MillisecondsInADay + Date.now(),
             data: true
         });
     }
 
-    private getOpenNotebookCount(): number {
-        const state = this.persistentState.createGlobalPersistentState<number>(DSSurveyStateKeys.OpenNotebookCount, 0);
-        return state.value;
+    private getOpenNotebookCount(type: BannerType): number {
+        let state: IPersistentState<number>;
+        switch (type) {
+            case BannerType.InsidersNotebookSurvey:
+                state = this.persistentState.createGlobalPersistentState<number>(
+                    InsidersNotebookSurveyStateKeys.OpenNotebookCount,
+                    0
+                );
+                return state.value;
+            case BannerType.ExperimentNotebookSurvey:
+                state = this.persistentState.createGlobalPersistentState<number>(
+                    ExperimentNotebookSurveyStateKeys.OpenNotebookCount,
+                    0
+                );
+                return state.value;
+            case BannerType.DSSurvey:
+            default:
+                state = this.persistentState.createGlobalPersistentState<number>(
+                    DSSurveyStateKeys.OpenNotebookCount,
+                    0
+                );
+                return state.value;
+        }
     }
 
-    private getExecutionCount(): number {
-        const state = this.persistentState.createGlobalPersistentState<number>(DSSurveyStateKeys.ExecutionCount, 0);
-        return state.value;
+    private getExecutionCount(type: BannerType): number {
+        let state: IPersistentState<number>;
+        switch (type) {
+            case BannerType.InsidersNotebookSurvey:
+                state = this.persistentState.createGlobalPersistentState<number>(
+                    InsidersNotebookSurveyStateKeys.ExecutionCount,
+                    0
+                );
+                return state.value;
+            case BannerType.ExperimentNotebookSurvey:
+                state = this.persistentState.createGlobalPersistentState<number>(
+                    ExperimentNotebookSurveyStateKeys.ExecutionCount,
+                    0
+                );
+                return state.value;
+            case BannerType.DSSurvey:
+            default:
+                state = this.persistentState.createGlobalPersistentState<number>(DSSurveyStateKeys.ExecutionCount, 0);
+                return state.value;
+        }
     }
 
     private async openedNotebook() {
-        const state = this.persistentState.createGlobalPersistentState<number>(DSSurveyStateKeys.OpenNotebookCount, 0);
-        await state.updateValue(state.value + 1);
-        return this.showBanner();
+        const state1 = this.persistentState.createGlobalPersistentState<number>(DSSurveyStateKeys.OpenNotebookCount, 0);
```

Comment:
This seems like you could call the same function with a different string 3 times to update the state for all 3.

```
private updateStateAndShowBanner(val: string, banner: BannerType) {
    const state = this.persistentState.createGlobalPersistentState<number>(val, 0);
    await state.updateValue(state.value + 1);
    this.showBanner(banner);
}

Reply:
yes, I'll also do that for getOpenNotebookCount and getExecutionCount